cdap-docs/developers-manual/source/building-blocks/transactional-messaging-system.rst

```diff
@@ -0,0 +1,287 @@
+.. meta::
+    :author: Cask Data, Inc.
+    :copyright: Copyright © 2017 Cask Data, Inc.
+    :description: Transactional Messaging System, a ACID-guaranteed "publish-and-subscribe" messaging service
+
+.. _transactional-messaging-system:
+
+==============================
+Transactional Messaging System
+==============================
+
+.. topic::  **Note: Beta Feature** 
+
+    TMS is currently a **beta** feature of CDAP |release|, and is subject to change without notice.
+
+
+Overview
+========
+The Transactional Messaging System (TMS) is a CDAP service that provides a
+"publish-and-subscribe" messaging system that understands transactions.
+
+It uses HBase for the persistent and durable storage of messages, and supports both
+transactional and non-transactional message publishing and consumption.
+
+TMS provides ACID *(atomicity, consistency, isolation,* and *durability)* guarantees,
+using `Apache Tephra <http://tephra.incubator.apache.org>`__ for the ACI portion. TMS
+doesn't interact directly with Tephra; instead, the client provides the transaction, as
+described below.
+
+TMS is similar to `Apache Kafka <https://kafka.apache.org>`__ but with added guarantees as
```

Comment:
Better not to mention Kafka in here as we don't want users to use TMS for transporting actual user data (which Kafka does), as the volume can be very high and TMS is not yet optimized for those use cases.